watcher.py

```diff
@@ -42,6 +53,8 @@ def execute(self, command):
             sys.stdout.flush()
             out += line
         proc.wait()
+        if not suppress and proc.returncode != 0:
+            raise subprocess.CalledProcessError(proc.returncode, command, out)
```

Comment:
We do - all of the git commands use the implicit value of suppress=False, meaning that errors encountered while running them will abort execution.